src/acme/pd/Company.java

```diff
@@ -1,211 +1,191 @@
+    @Column(name = "COURIER_MILES_PER_HOUR")
+    private double courierMilesPerHour = 5.8;
+    @Transient
+    private String mapFile = "map/map.csv";
+
     public Company() {
-        // TODO initialize company
-        File file = new File(mapFile);
-        this.map = new Map(file);
-
-        couriers = new HashMap<UUID, Courier>();
-        tickets = new HashMap<UUID, Ticket>();
-        customer = new HashMap<UUID, Customer>();
+        // TODO initialize company
+        File file = new File(mapFile);
+        this.map = new Map(file);
+
+        couriers = new HashMap<UUID, Courier>();
+        tickets = new HashMap<UUID, Ticket>();
+        customers = new HashMap<UUID, Customer>();
+        currentUser = new User();
         users = new HashMap<UUID, User>();
-
-        // TODO remove test customers and couriers
-        for (int i = 0; i < 10; i++) {
-            Courier c1 = new Courier();
-            c1.setName("That Guy " + i);
-            couriers.put(UUID.randomUUID(), c1);
-            Customer c2 = new Customer();
-            c2.setName("That Customer " + (1000 + i));
-            customer.put(UUID.randomUUID(), c2);
-
-            User u1 = new User();
-            u1.setName("First " + "Last" + i);
-            u1.setUsername("uname " + i);
-            if (i % 2 == 0) {
-                u1.setActive(true);
-            } else {
-                u1.setActive(false);
-            }
-            u1.setAdmin(false);
-            u1.setPassword("pass"+i);
-            users.put(UUID.randomUUID(), u1);
-        }
     }
-
+
     public UUID getId() {
-        // TODO fix with database
+        // TODO fix with database
         /*
          * if (this.id == null) { this.id = UUID.randomUUID(); }
          */
-        return id;
-    }
-
+        return id;
+    }
+
     public String getName() {
-        return name;
-    }
-
+        return name;
+    }
+
     public void setName(String name) {
-        this.name = name;
-    }
-
+        this.name = name;
+    }
+
     public Map getMap() {
-        return map;
-    }
-
+        return map;
+    }
+
     public void setMap(Map map) {
-        this.map = map;
-    }
-
-    public HashMap<UUID, Courier> getCouriers() {
-        return couriers;
-    }
-
-    public void setCouriers(HashMap<UUID, Courier> couriers) {
-        this.couriers = couriers;
-    }
-
-    public HashMap<UUID, Ticket> getTickets() {
-        return tickets;
-    }
-
-    public void setTickets(HashMap<UUID, Ticket> tickets) {
-        this.tickets = tickets;
-    }
-
-    public HashMap<UUID, Customer> getCustomer() {
-        return customer;
-    }
-
-    public void setCustomer(HashMap<UUID, Customer> customer) {
-        this.customer = customer;
-    }
-
-    public HashMap<UUID, User> getUsers() {
+        this.map = map;
+    }
+
+    public java.util.Map<UUID, Courier> getCouriers() {
+        return couriers;
+    }
+    
+    public void addCourier(Courier courier) {
```

Comment:
Does this auto add them to the database or do we need to save them first?

Reply:
No, `PersistableEntity` interface (that `Ticket` implements) has methods that handle database interactions.

Reply:
Then doesn't hibernate automatically handle updating the list on a create?

Reply:
Hibernate handles the association between `Company` and its collection of `couriers`, but the courier must first be saved to the database solely for the reason that `Company.couriers` is a map with `courier.id` and not just a list of couriers.